src/components/search/Suggestions/Suggestions.tsx

```diff
@@ -21,28 +27,37 @@ const SearchSuggestionsQuery = gql`
   }
 `
 
-function useSuggestions(term: string) {
+function useSuggestions(term: string, limit: number = MAX_SUGGESTIONS) {
   const [suggestions, setSuggestions] =
     useState<SearchSuggestionsQueryQuery['search']['suggestions']>()
 
   const [loading, setLoading] = useState<boolean>(false)
 
-  useEffect(() => {
-    if (term.length > 0) {
-      setLoading(true)
-      request<
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedRequest = useCallback(
```

Comment:
can't we use React18 features like useDefferedValue to not need to implement a debounce function?

Reply:
I'm going to check that function out! Thanks.